love_letter_cards.py

```diff
@@ -123,5 +125,11 @@ class Princess(Card):
     def __init__(self):
         super().__init__("Princess", 8)
 
-    def action(self, player, other_card_in_hand):
+    def action(self, other_player):
         print("do Princess action")
+
+class InvalidActionError(Exception):
+    ''' Exception that is thrown when a player attempts an invalid action such as attacking another player protected by Handmaid.'''    
+    def __init__(self, message):
+        self.errorMessage = message
```

Comment:
Interesting.  I hadn't thought of that - mainly going down the "we won't pick invalid actions" path.  BUT this makes more sense when we get to the point where there are different ways to select the card to play and the opponent to play against.  Nice!